src/app/games/conspirateurs/ConspirateursMove.ts

```diff
@@ -73,8 +61,8 @@ export class ConspirateursMoveSimple extends MoveCoordToCoord {
     public toString(): string {
         return `ConspirateursMoveSimple(${this.getStart().toString()} -> ${this.getEnd().toString()})`;
     }
-    public equals(other: ConspirateursMove): boolean {
-        if (other.isSimple()) {
+    public override equals(other: ConspirateursMove): boolean {
+        if (ConspirateursMove.isSimple(other)) {
```

Comment:
cette branche peut simplement appeler super.equals non ?

Reply:
certes! mais il a fallu un as this

Reply:
c'est ben bizarre ça, t'as quoi comme erreur sinon ? Car il doit savoir que dans le if, `other` est un `ConspirateursMoveSimple` avant tout, et puis bah `super.equals` c'est définit sur `MoveCoord`, et `ConspirateursMoveSimple` est bien un `MoveCoord`

Reply:
```
Argument of type 'ConspirateursMoveSimple' is not assignable to parameter of type 'this'.
  'ConspirateursMoveSimple' is assignable to the constraint of type 'this', but 'this' could be instantiated with a different subtype of constraint 'ConspirateursMoveSimple'.
```

Reply:
ah! Plutôt que `ConspirateursMove.isSimple(other)`, fait: `other instanceof this` !

Reply:
ah oui c'est le `this` dans le super qui pose problème, tu peux faire ceci :
```
    public override equals(other: this): boolean {
        if (ConspirateursMove.isSimple(other)) {
            return super.equals(other);
        } else {
            return false;
        }
    }
```
Et adapter similairement pour le drop, car là on rely pas sur super dedans, il faudrait.

Reply:
Si on modifie ça, on sais plus faire simpleMove.equals(notASimpleMove) et du coup on ne sais pas faire move.equals(move), c'qu'est pas souhaitable je dirais.

Reply:
aah mais c'est parce que super.equals attend un this, donc notre typage est tout foiré ! Tu peux laisser le `as` du coup et résoudre